rust-symcrypt/src/hash.rs

```diff
@@ -0,0 +1,186 @@
+use core::ffi::c_void;
+use std::mem;
+use std::ptr;
+use symcrypt_sys;
+
+pub const SHA256_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
+pub const SHA384_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
+
+pub struct Sha256State {
+    state: symcrypt_sys::SYMCRYPT_SHA256_STATE,
+}
+
+impl Sha256State {
+    pub fn new() -> Self {
+        let mut instance = Sha256State {
+            state: symcrypt_sys::SYMCRYPT_SHA256_STATE::default(),
+        };
+        unsafe {
+            symcrypt_sys::SymCryptSha256Init(&mut instance.state);
+        }
+        instance
+    }
+
+    pub fn append(&mut self, data: &[u8]) {
+        unsafe {
+            symcrypt_sys::SymCryptSha256Append(
+                &mut self.state,                    // pState
```

Comment:
If you use rust-analyzer, does the `pState: ` inlay show up? If so, it's probably okay to leave off the `// pState` part, although that's a project style question.

Reply:
I use rust analyzer which does show the pState: But I wanted to keep the comments for clarity, but if majority of users have some sort of rust analyzer extension I will remove the // pState comments.

Reply:
As far as I know, the vast majority of Rust users use rust-analyzer. All the ones I know personally do, but I don't have any kind of representative survey data to say one way or the other.

Reply:
I feel that it would be pretty common. I will remove the comments